src/Features/Core/Portable/Diagnostics/EngineV2/DiagnosticIncrementalAnalyzer.ProjectState.cs

```diff
@@ -47,15 +47,15 @@ public bool IsEmpty(DocumentId documentId)
             /// <summary>
             /// Return all diagnostics for the given project stored in this state
             /// </summary>
-            public async Task<DiagnosticAnalysisResult> GetAnalysisDataAsync(IPersistentStorageService persistentService, Project project, bool avoidLoadingData, CancellationToken cancellationToken)
+            public async Task<DiagnosticAnalysisResult> GetAnalysisDataAsync(Project project, bool avoidLoadingData, CancellationToken cancellationToken)
             {
                 // make a copy of last result.
                 var lastResult = _lastResult;
                 Contract.ThrowIfFalse(lastResult.ProjectId == project.Id);
 
                 if (lastResult.IsDefault)
                 {
-                    return await LoadInitialAnalysisDataAsync(persistentService, project, cancellationToken).ConfigureAwait(false);
+                    return await LoadInitialAnalysisDataAsync(project, cancellationToken).ConfigureAwait(false);
```

Comment:
it's unclear to me if this even makes sense.  what are we loading initial data from?  that said, i'm not sure the lifetimes of proejct-states, so it's possible this has meaning otuside of the concept of persistence across sessions.